src/osgEarth/VirtualProgram.cpp

```diff
@@ -291,13 +291,19 @@ namespace
     }
 
 
-    void parseShaderForMerging( const std::string& source, unsigned& version, std::string& subversion, HeaderMap& precisions, HeaderMap& headers, std::stringstream& body )
```

Comment:
Down the road we should break the shader merging code out into ShaderUtils or something.

Reply:
100% now osg has the condintional compilation of shaders the merging has become way more complicated. This will trip over in certain cases. I’ve informed chris that a rewrite is a better approach then hacking a fix everytime a shadet combination fails to compile due to bad merging.